soh/src/code/z_parameter.c

```diff
@@ -1415,15 +1437,18 @@ void Inventory_SwapAgeEquipment(void) {
         }
 
         // When becoming adult, remove swordless flag since we'll get master sword
+        // (Unless Master Sword is shuffled)
         // Only in rando to keep swordless link bugs in vanilla
-        if (gSaveContext.n64ddFlag) {
+        if (gSaveContext.n64ddFlag && !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
             gSaveContext.infTable[29] &= ~1;
         }
 
         gSaveContext.childEquips.equipment = gSaveContext.equips.equipment;
 
         if (gSaveContext.adultEquips.buttonItems[0] == ITEM_NONE) {
-            gSaveContext.equips.buttonItems[0] = ITEM_SWORD_MASTER;
+            if (!gSaveContext.n64ddFlag || !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
+                gSaveContext.equips.buttonItems[0] = ITEM_SWORD_MASTER;
+            }
```

Comment:
```suggestion
            if (!gSaveContext.n64ddFlag || !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
                gSaveContext.equips.buttonItems[0] = ITEM_SWORD_MASTER;
            } else {
                gSaveContext.equips.buttonItems[0] = ITEM_NONE;
            }
```

I believe this is needed otherwise if child link has kokiri sword equipped then adult link will have kokiri sword